### PR TITLE
Undo/Redo of structures

### LINKED ISSUE #142: Undo/Redo structures
As a text writer that likes to play around with different designs, I want to able to undo and redo structures that I applied to the text, in order to improve productivity and ease of use of the structures.

Acceptance criteria:
- [ ] When applying a structure to a selection and pressing the `Undo` button, the structure is no longer applied to the text.
- [ ] When applying a structure to a selection, pressing the `Undo` button and the `Redo` button afterwards, the structure is again applied to the text
- [ ] The functionality stated above works with shortcuts and the right click menu

Estimated effort: `8`

### PR DESCRIPTION
**Pull request by sirkrypt0**
Closes #142

### CODE REVIEW

**Comment by frcroth** [comment]
add tests

**Comment by sirkrypt0** [comment]
Will resolve conflicts after review

**Review by frcroth — APPROVED** [review]
I'm not 100% satisfied with the naming of changeAttributesFor: with: but other than it looks good

File: packages/RichTextEditing-Tests.package/RichTextEditorTests.class/instance/testRedoStructureUnapplyUnappliesStructureWith..st [hunk 1]
```diff
@@ -0,0 +1,9 @@
+tests
+testRedoStructureUnapplyUnappliesStructureWith: aBlock
+
+	self 
+		setupBoldText;
+		deselectOptionBold.
+	self editorModel undo.
+	aBlock value.
```
Comment (j-hellenberg) [inline packages/RichTextEditing-Tests.package/RichTextEditorTests.class/instance/testRedoStructureUnapplyUnappliesStructureWith..st]: Perhaps name it `redoBlock` as you did in `testRedoStructureWith` to make clear what the block is for

File: packages/RichTextEditing-Tests.package/RichTextEditorTests.class/instance/testUndoStructureUnapplyReappliesStructureWith..st [hunk 2]
```diff
@@ -0,0 +1,8 @@
+tests
+testUndoStructureUnapplyReappliesStructureWith: aBlock
```
Comment (j-hellenberg) [inline packages/RichTextEditing-Tests.package/RichTextEditorTests.class/instance/testUndoStructureUnapplyReappliesStructureWith..st]: Same here: What is this block supposed to do?

**Review by j-hellenberg — COMMENTED** [review]
That's a lot of clean tests :)

File: packages/RichTextEditing-Tests.package/RichTextEditorTests.class/instance/testRedoStructureUnapplyUnappliesStructureWith..st (hunk 1 above)
Comment (j-hellenberg) [reply packages/RichTextEditing-Tests.package/RichTextEditorTests.class/instance/testRedoStructureUnapplyUnappliesStructureWith..st]: Does this really belong into the tests category as it is only an utilily for acceptance tests as far as I understood?

**Review by frcroth — APPROVED** [review]

**Review by j-hellenberg — APPROVED** [review]

**Pull request merged**